src/AI/CapPhaseAnalyzer.java

```diff
@@ -0,0 +1,454 @@
+    while (!rightfulFactories.isEmpty())
+    {
+      final XYCoord dest = rightfulFactories.remove(0);
+      Utils.sortLocationsByDistance(dest, startingFactories);
+      final XYCoord start = startingFactories.get(0);
+      final Commander owner = factoryOwnership.get(start);
+
+      final Unit inf = new Unit(owner, owner.getUnitModel(UnitModel.TROOP));
+      inf.x = start.xCoord;
+      inf.y = start.yCoord;
+
+      final GamePath infPath = findFeasiblePath(inf, dest, map);
+      if( null == infPath || infPath.getPathLength() < 1 )
+        continue; // Can't reach
+
+      int distance = infPath.getFuelCost(inf, map);
+
+      ArrayList<CapStop> chain = new ArrayList<>();
+      CapStop build = new CapStop(start);
+      build.extraTurns = distance/inf.getMovePower(map) - 13;
+      chain.add(build);
+      CapStop cap = new CapStop(dest);
+      chain.add(cap);
+      factoryCapChains.add(chain);
+
+      // Now that we're just in cap-chain land, don't worry about whether we start with this factory or not.
+      startingFactories.add(dest);
+    }
+
+    buildBaseCapChains(map, viewer, rightfulProps, startingFactories);
+
+    // Add our factory chains in at the start of each list
+    for(ArrayList<CapStop> chain : factoryCapChains)
+    {
+      XYCoord start = chain.get(0).coord;
+      capChains.get(start).add(0, chain);
+    }
+  }
+
+  public void buildBaseCapChains(GameMap map, Army viewer, ArrayList<XYCoord> rightfulProps, ArrayList<XYCoord> startingFactories)
+  {
+    ArrayList<XYCoord> remainingFactories = new ArrayList<>(startingFactories);
+    // Build initial bits of capChains
+    for( XYCoord start : remainingFactories )
+    {
+      ArrayList<ArrayList<CapStop>> chainList = new ArrayList<>();
+      capChains.put(start, chainList);
+    }
+
+    boolean madeProgress = true;
+    final Unit inf = new Unit(viewer.cos[0], viewer.cos[0].getUnitModel(UnitModel.TROOP));
+    if( startingFactories.size() > 0 )
+    {
+      inf.x = startingFactories.get(0).xCoord;
+      inf.y = startingFactories.get(0).yCoord;
+    }
+    final int infMove = inf.getMovePower(map);
+
+    // Find the next stop or iterate extraTurns on all cap chains
+    while (madeProgress && !rightfulProps.isEmpty())
+    {
+      // Create new cap chains
+      for( XYCoord start : remainingFactories )
+      {
+        ArrayList<CapStop> chain = new ArrayList<>();
+        CapStop build = new CapStop(start);
+        chain.add(build);
+        capChains.get(start).add(0, chain);
+      }
+
+      madeProgress = false;
+      for( ArrayList<ArrayList<CapStop>> chainList : capChains.values() )
+        for( ArrayList<CapStop> chain : chainList )
+        {
+          if( rightfulProps.isEmpty() )
+            break;
+
+          CapStop last = chain.get(chain.size() - 1);
+          if( last.extraTurns >= LOOKAHEAD_TURNS )
+          {
+            if (chain.size() == 1)
+              remainingFactories.remove(last.coord);
```

Comment:
This condition indicates that the factory at `last.coord` can't reach anything to capture within `LOOKAHEAD_TURNS`, so we just remove it from further consideration?

Reply:
Yeah. That's the "base case" if this were recursive.
We've determined that all the cap chains we *can* make from that factory have been made already, so there's no point spinning on it anymore.